pushkin/sender/senders.py

```diff
@@ -44,6 +45,40 @@ def log_notifications(self, notifications):
             except:
                 main_logger.exception("Error while logging notification to csv log!")
 
+class NotificationOperation():
+    '''Represents operation which should be executed outside of process pool'''
+    def __init__(self, operation, data):
+        self.operation = operation
+        self.data = data
+
+class NotificationPostProcessor(Thread):
+    UPDATE_CANONICALS = 1
+    UPDATE_UNREGISTERED_DEVICES = 2
+    OPERATION_QUEUE = multiprocessing.Queue()
+
+    def __init__(self):
+        Thread.__init__(self)
+        self.daemon = True
+
+    def update_canonicals(self, canonical_ids):
+        database.update_canonicals(canonical_ids)
+
+    def update_unregistered_devices(self, unregistered_devices):
+        database.update_unregistered_devices(unregistered_devices)
+
+    def run(self):
+        while True:
+            try:
+                record = NotificationPostProcessor.OPERATION_QUEUE.get()
+                if record.operation == NotificationPostProcessor.UPDATE_CANONICALS:
```

Comment:
I think it would be cleaner to solve this with polymorphism instead with if. If we say we require NotificationPostProcessor tasks to have execute method, you could avoid mapping/if here. That also means we can add more tasks without having to change this class.